src/main/java/com/amazon/ion/system/IonReaderBuilder.java

```diff
@@ -263,6 +269,77 @@ public IonBufferConfiguration getBufferConfiguration() {
+     *     called at least once on this builder instance, OR</li>
+     *     <li>any stream interceptors detected on the classpath using
+     *     {@link ServiceLoader#load(Class)}, if this method was not called on this builder
+     *     instance.</li>
+     * </ol>
+     *
+     * @param streamInterceptor the stream interceptor to add.
+     *
+     * @return this builder instance, if mutable;
+     * otherwise a mutable copy of this builder.
+     */
+    public IonReaderBuilder addInputStreamInterceptor(InputStreamInterceptor streamInterceptor) {
+        IonReaderBuilder b = mutable();
+        if (b.streamInterceptors == null) {
+            b.streamInterceptors = new ArrayList<>(DEFAULT_STREAM_INTERCEPTORS);
+        }
+        b.streamInterceptors.add(streamInterceptor);
+        return b;
+    }
+
+    /**
+     * Detects implementations of {@link InputStreamInterceptor} available to the {@link ClassLoader} that loaded
+     * this class, appending any implementations found to the list of stream interceptors enabled by default.
+     * @return the stream interceptors.
+     */
+    private static List<InputStreamInterceptor> detectStreamInterceptorsOnClasspath() {
+        ServiceLoader<InputStreamInterceptor> loader = ServiceLoader.load(
+            InputStreamInterceptor.class,
+            // The ClassLoader used to load this class. Each ClassLoader may have access to different resources.
+            IonReaderBuilder.class.getClassLoader()
+        );
+        Iterator<InputStreamInterceptor> interceptorIterator = loader.iterator();
+        if (!interceptorIterator.hasNext()) {
+            // Avoid allocating a new list in the common case: no custom interceptors detected.
+            return DEFAULT_STREAM_INTERCEPTORS;
+        }
+        List<InputStreamInterceptor> interceptorsOnClasspath = new ArrayList<>(4); // 4 is arbitrary, but more would be very rare.
+        interceptorsOnClasspath.addAll(DEFAULT_STREAM_INTERCEPTORS);
+        interceptorIterator.forEachRemaining(interceptorsOnClasspath::add);
+        return interceptorsOnClasspath;
+    }
+
+    /**
+     * Gets the {@link InputStreamInterceptor} instances available to this builder. If any instances were added using
+     * {@link #addInputStreamInterceptor(InputStreamInterceptor)}, then the returned list will be the default stream
+     * interceptor, which detects GZIP, followed by the stream interceptor(s) manually added. If no instances were
+     * manually added, the returned list will be the default stream interceptor followed by any stream interceptor(s)
+     * detected on the classpath by {@link ServiceLoader#load(Class)}.
+     * @see #addInputStreamInterceptor(InputStreamInterceptor)
+     * @return an unmodifiable view of the stream interceptors currently configured.
+     */
+    public List<InputStreamInterceptor> getInputStreamInterceptors() {
+        if (streamInterceptors == null) {
+            return DETECTED_STREAM_INTERCEPTORS;
```

Comment:
Spotbugs warns here:
> EI_EXPOSE_REP: May expose internal representation by returning reference to mutable object

However, this list is unmodifiable. Please double-check this in your review. Rather than do something redundant to satisfy Spotbugs, I decided to re-baseline (effectively ignoring this violation).